src/clock.rs

```diff
@@ -133,11 +162,17 @@ impl ClockSetup {
         let core_pll = PLLConfig::calculate(HFXCLK, coreclk).unwrap();
         let hfpclk_pll = PLLConfig::calculate(HFXCLK, pclk * 2).unwrap();
 
+        // Remember `hfxclk`'s current setting and restore on exit; enable, if necessary
+        let _guard = HfxclkGuard::new(&self.prci);
+        if self.prci.hfxosccfg.read().hfxoscen().bit_is_clear() {
+            self.prci.hfxosccfg.modify(|_, w| w.hfxoscen().set_bit());
```

Comment:
This bit turns oscillator on, but the chip is able to work without a quartz, with hfxclkin. What is the reason to force oscillator here? We don't have software control over hfxsel anyway.

Reply:
Because on l.175, we switch the core clock to it; I don't know what would happen if we switched to a disabled oscillator...

Reply:
No, we switch to hfclk, it comes out of a mux between the oscillator and hfxclkin. If hfxclkin is selected, we don't need the oscillator at all.

Reply:
Got it--I hadn't realized that until your first comment, thanks!